### PR TITLE
Merge flake8 configurations

### PR DESCRIPTION
**Pull request by krishnakalyan3**
#1053

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [flake8]
 max-line-length = 120
-ignore = E305,E402,E721,E741,F401,F403,F405,F999,W503,W504
-exclude = build,docs/source,_ext,third_party
+ignore = E305,E402,E721,E741,F401,F403,F405,W503,W504,F821,F841,F999
+exclude = build,docs/source,_ext,third_party,docs/src,venv,torch/lib/gloo,torch/lib/pybind11,torch/lib/nanopb
```
Comment (mthrok) [inline .flake8:4]: Let's remove unused ones. Only `build,docs/source,third_party` are relevant and the others are originated from PyTorch core.

File: .flake8 [hunk 2]
```diff
@@ -1,4 +1,4 @@
 [flake8]
 max-line-length = 120
-ignore = E305,E402,E721,E741,F401,F403,F405,F999,W503,W504
-exclude = build,docs/source,_ext,third_party
+ignore = E305,E402,E721,E741,F401,F403,F405,W503,W504,F821,F841,F999
```
Comment (mthrok) [inline .flake8:3]: What happens if `F821` and `F841` are not added here? They should not be added here I think.

Comment (mthrok) [reply .flake8:3]: Can you also see what happens if removing 401, 402, 403, 405, 503, 504? I do not see a good reason in torchaudio that they should be ignored.

Comment (krishnakalyan3) [reply .flake8:3]: ref https://github.com/pytorch/pytorch/blob/master/.flake8#L7

Comment (krishnakalyan3) [reply .flake8:3]: I agree that we should remove `F821,  F841`

Comment (krishnakalyan3) [reply .flake8:3]: `F503, F504` looks like its being ignore in pytorch too.

https://www.flake8rules.com/rules/W504.html
https://www.flake8rules.com/rules/W503.html

From what I understand they seem to be contradicting each other

Comment (krishnakalyan3) [reply .flake8:3]: `403,405` being ignore by pytorch too. (Similar rules discouraging the use for * )


https://www.flake8rules.com/rules/F403.html
https://www.flake8rules.com/rules/F405.html

Comment (krishnakalyan3) [reply .flake8]: image

 looks like we have `401` violations.

Comment (mthrok) [reply .flake8]: Would it possible for you to fix it?

For `__init__.py`s, listing the imported stuff in `__all__` would do. https://stackoverflow.com/a/31079085
For others, unused import should be removed.

If that uncovers other issues, (like unused imported function is used from external module). We can just give up and do it in a follow up.

**Comment by mthrok** [comment]
@krishnakalyan3 I am merging this one now. If you would like to follow up on the change to the rule. Feel free to open a new PR.

**Comment by mthrok** [comment]
Thanks!

**Pull request merged**